### PR TITLE
feat: migrate angular to version 15

### PR DESCRIPTION
**Pull request by mavliut-fozliev**

### CODE REVIEW

File: apps/spica/environments/environment.ts [hunk 1]
```diff
@@ -5,5 +5,5 @@
 
 export const environment = {
   production: true,
-  api: "http://localhost:4300"
```
Comment (htuna07) [inline apps/spica/environments/environment.ts:8]: Could you revert this?

File: apps/spica/src/styles.scss [hunk 2]
```diff
@@ -86,19 +98,19 @@ mat-card {
...
     }
   }
-  .mat-form-field-label {
+  .mdc-form-field__label {
     top: 2.14375em !important;
   }
-  &.mat-form-field-appearance-outline.mat-form-field-can-float.mat-form-field-should-float {
-    .mat-form-field-label {
+  &.mdc-form-field--outlined.mdc-form-field--can-float.mdc-form-field--should-float {
+    .mdc-form-field__label {
```
Comment (htuna07) [inline apps/spica/src/styles.scss:113]: Could you explain or refer to why these class names include kebab and snake case naming? Also, some of them include double dashes between words, are they running as expected?

**Review by htuna07 — COMMENTED** [review]
Needs clarification

File: apps/spica/src/styles.scss (hunk 2 above)
Comment (mavliut-fozliev) [reply apps/spica/src/styles.scss:113]: This is the result of the migration script. 
Screenshot 2024-12-31 at 11 06 59

**Review by htuna07 — APPROVED** [review]

**Pull request merged**